Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp

```diff
@@ -2184,33 +2188,36 @@ void NetworkResourceLoader::useRedirectionForCurrentNavigation(WebCore::Resource
 
 void NetworkResourceLoader::sendDidReceiveDataMessage(const FragmentedSharedBuffer& buffer, size_t encodedDataLength)
 {
-    auto delta = bytesTransferredOverNetworkDelta();
+    RefPtr networkLoad = m_networkLoad;
+    auto bytesTransferredOverNetwork = networkLoad ? networkLoad->bytesTransferredOverNetwork() : 0;
 
-    send(Messages::WebResourceLoader::DidReceiveData(IPC::SharedBufferReference(buffer), encodedDataLength, delta));
+    updateBytesTransferredOverNetwork(bytesTransferredOverNetwork);
+
+    send(Messages::WebResourceLoader::DidReceiveData(IPC::SharedBufferReference(buffer), encodedDataLength, bytesTransferredOverNetwork));
+}
+
+void NetworkResourceLoader::updateBytesTransferredOverNetwork(size_t bytesTransferredOverNetwork)
+{
+    CheckedSize delta = bytesTransferredOverNetwork - m_bytesTransferredOverNetwork;
+    ASSERT(!delta.hasOverflowed());
+    m_bytesTransferredOverNetwork = bytesTransferredOverNetwork;
```

Comment:
This seems like it shouldn't be necessary anymore? It looks like you changed the workers code in WebProcess to deal with a total amount rather than a delta.

Reply:
On the older network stack and non-Cocoa platforms, the size is always zero. So still checking the delta to reduce unnecessary IPC for shared workers.

Reply:
In that case, maybe you would want to rename `m_bytesTransferredOverNetwork` to `m_bytesTransferredOverNetworkSinceLastSharedWorkerReport` or something.

And also isn't the line on 2209 incorrect then? It seems like it should be passing along `bytesTransferredOverNetwork` not `m_bytesTransferredOverNetwork`.